File: c/space-age/src/space_age.c
#include "space_age.h"

float convert_planet_age(planet_t planet, int64_t input) {
    switch (planet) {
        case EARTH:
            return input / 31557600.0;
        case MERCURY:
            return convert_planet_age(EARTH, input) / 0.2408467;
        case VENUS:
            return convert_planet_age(EARTH, input) / 0.61519726;
        case MARS:
            return convert_planet_age(EARTH, input) / 1.8808158;
        case JUPITER:
            return convert_planet_age(EARTH, input) / 11.862615;
        case SATURN:
            return convert_planet_age(EARTH, input) / 29.447498;
        case URANUS:
            return convert_planet_age(EARTH, input) / 84.016846;
        case NEPTUNE:
            return convert_planet_age(EARTH, input) / 164.79132;
        default:
            return 0.0;
    }
}

File: javascript/space-age/space-age.js
const EARTH_YEAR_SECOND = 3.15576e7;

const ORBITAL_PERIODS = {
    earth: 1,
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
}
const FORMAT = new Intl.NumberFormat('en-IN', { maximumSignificantDigits: 2 });

export const age = (planet, seconds) => {
    const ageOnEarth = seconds / EARTH_YEAR_SECOND;
    const ageOnPlanet = ageOnEarth / ORBITAL_PERIODS[planet];
    return FORMAT.format(ageOnPlanet);
};

function round(number) {
    return Math.round(number * 100) / 100;
}
